### PR TITLE
implement DLR loss in TF2

### PR DESCRIPTION
**Pull request by CNOCycle**
Please read the more details in https://github.com/fra31/auto-attack/issues/37

### CODE REVIEW

File: autoattack/utils_tf2.py [hunk 1]
```diff
@@ -193,11 +193,27 @@ def get_logits_loss_grad_target(self, x, y, y_target):
         
         return torch.from_numpy(logits_val.numpy()).cuda(), torch.from_numpy(loss_indiv_val.numpy()).cuda(), torch.from_numpy(grad_val.numpy()).cuda()
 
-def dlr_loss(x, y, num_classes=10):
-    x_sort = tf.sort(x, axis=1)
-    y_onehot = tf.one_hot(y, num_classes)
-    ### TODO: adapt to the case when the point is already misclassified
-    loss = -(x_sort[:, -1] - x_sort[:, -2]) / (x_sort[:, -1] - x_sort[:, -3] + 1e-12)
+def dir_loss(x, y, num_classes=10):
```
Comment (fra31) [inline autoattack/utils_tf2.py:196]: Please rename to `dlr_loss`.

**Comment by fra31** [comment]
Beside the minor comment above, I think we can merge it.

**Comment by fra31** [comment]
Thanks again for the contribution!
